test/e2e/terminal.test.ts

```diff
@@ -0,0 +1,62 @@
+import { test, expect } from "@playwright/test"
+import * as fs from "fs"
+import { tmpdir } from "os"
+import * as path from "path"
+
+import { STORAGE } from "../utils/constants"
+import { CodeServer } from "./models/CodeServer"
+
+test.describe("Integrated Terminal", () => {
+  // Create a new context with the saved storage state
+  // so we don't have to logged in
+  const options: any = {}
+  const testFileName = "test.txt"
+  const testString = "new string test from e2e test"
+  let codeServer: CodeServer
+
+  // TODO@jsjoeio
+  // Fix this once https://github.com/microsoft/playwright-test/issues/240
+  // is fixed
+  if (STORAGE) {
+    const storageState = JSON.parse(STORAGE) || {}
+    options.contextOptions = {
+      storageState,
+    }
+  }
+  test.beforeEach(async ({ page }) => {
+    codeServer = new CodeServer(page)
+    await codeServer.navigate()
+  })
+
+  test("should echo a string to a file", options, async ({ page }) => {
+    const tmpFolderPath = fs.mkdtempSync(path.join(tmpdir(), "code-server-test"))
```

Comment:
Note: I tried to use `tmpdir` from `src/node/util` but ran into this error. I think it's because Playwright requires the test files and something doesn't support enums 

![image](https://user-images.githubusercontent.com/3806031/115480829-2d618e80-a200-11eb-844a-2b8d86dbd98c.png)

Leaving as is.

Reply:
haha you might want to add a comment with the stack trace etc, so people don't spend time trying the same thing later :stuck_out_tongue:

Reply:
Maybe we can move it to `src/node/constants.ts`. Or if that fails we can duplicate it for now in `test/utils/constants.ts`.

Reply:
Good idea! I'll try that.

Reply:
I moved it to `constants` but new error :( 

I think Playwright uses CommonJS and we're using ES Modules so anything that isn't done the "CommonJS" way is going to cause Playwright to throw.

![image](https://user-images.githubusercontent.com/3806031/115616261-da8de280-a2a4-11eb-946e-5efa50d2f3e1.png)

I think the refactor still makes sense, so I'll leave it, but I'll leave the test how it is and leave a comment.

Reply:
Possibly related? https://github.com/microsoft/playwright/issues/2269#issue-619578701

Reply:
Maybe the `"esModuleInterop": true` option in our `tsconfig.json` is allowing ES Module-style imports and we should remove that.

Reply:
Ah, maybe you're right: [docs](https://www.typescriptlang.org/tsconfig#esModuleInterop).

I wonder if turning that off will break anything.